torch/nn/modules/loss.py

```diff
@@ -690,8 +690,10 @@ class BCEWithLogitsLoss(_Loss):
             elements in the output, ``'sum'``: the output will be summed. Note: :attr:`size_average`
             and :attr:`reduce` are in the process of being deprecated, and in the meantime,
             specifying either of those two args will override :attr:`reduction`. Default: ``'mean'``
-        pos_weight (Tensor, optional): a weight of positive examples.
-                Must be a vector with length equal to the number of classes.
+        pos_weight (Tensor, optional): a weight of positive examples. Must be a tensor with equal 
+            dimensions to the target tensor and equal size along the class dimension to the number 
+            of classes. To apply the same positive weight along all spacial dimensions for a 2D 
```

Comment:
To clarify, must the number of dimensions be equal to that of the target tensor? My understanding is that there are two separate cases here

1) the case mentioned in #65702 where for target of size `[B, C, H, W]` we can either have pos_weight of size `[B, C, H, W]` (where B is batch size) so we apply different pos weights to each element of the batch or `[C, H, W]` (same pos weights across batch)

2) the case mentioned in #82496 where for multidimensional labels `pos_weight` must necessarily have as many dimensions as the label in order to be broadcasted across the spacial dimensions

I think both these cases are in line with the expected [broadcasting semantics](https://pytorch.org/docs/stable/notes/broadcasting.html). Perhaps it makes sense to say that `pos_weight` is broadcasted with target, but draw attention to the broadcasting semantics documentation

Reply:
Ah yes, you are correct. I have updated to PR to reflect this more accurately with the examples. I find the case in https://github.com/pytorch/pytorch/issues/65702 interesting since 'weight' seems to be the intended parameter to manage weighing over batches, although not just for positive examples.

Reply:
Hm, good catch! Looking at the implementation though, it seems like we first [multiply (and hence broadcast) `pos_weight` with `target`](https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/native/Loss.cpp#L366-L368) and then [later multiply with weight](https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/native/Loss.cpp#L374)